Tests/test_image.py

```diff
@@ -193,6 +193,10 @@ def test_internals(self):
         assert not im.readonly
 
     @pytest.mark.skipif(is_win32(), reason="Test requires opening tempfile twice")
+    @pytest.mark.skipif(
+        sys.platform == "cygwin",
+        reason="Test requires opening an mmaped file for writing",
+    )
```

Comment:
Cygwin can't use Python's mmap? Could you provide some more information about this, or link to a discussion about it?

Reply:
```bash
$ echo "Lorem itsum dolor sit amet" > a.txt
$ python
```
```python
>>> import mmap
>>> with open("a.txt", "r") as in_file:
...     with mmap.mmap(in_file.fileno(), 30, access=mmap.ACCESS_COPY) as mapped:
...         with open("a.txt", "w") as out_file:
...             out_file.write(mapped)
...
Traceback (most recent call last):
  File "<stdin>", line 3, in <module>
PermissionError: [Errno 13] Permission denied: 'a.txt'
```
My initial comment was based on the above test case failing.  In a fresh python session, the below case works fine:
```python
>>> with open("a.txt", "r") as in_file:
...     with open("a.txt", "w") as out_file:
...         out_file.write(in_file.read())
...
0
```
so it looks like having the mmap open while opening the file for writing is a problem, but opening the mmap seems to be fine.  Since the test case is nearly this exact sequence of operations, I marked it as XFAIL.

I get a different error using Windows-native python from anaconda:
```python
>>> import mmap
>>> with open("a.txt", "r") as in_file:
...     with mmap.mmap(in_file.fileno(), 30, access=mmap.ACCESS_COPY) as mapped:
...         with open("a.txt", "w") as out_file:
...             out_file.write(mapped)
...
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
OSError: [WinError 8] Not enough memory resources are available to process this command
```
I have no idea why that error is happening; I have gigabytes of memory free and requested 30 bytes.